File: Fundamentals of Discrete mathematics/Java/task_1.java
import java.util.*;

class SetTransform {
    public static Set<Integer> inputSet() {
        try (Scanner scanner = new Scanner(System.in)) {
            while (true) {
                String userInput = scanner.nextLine().trim();
                String[] inputList = userInput.split(" ");
                try {
                    Set<Integer> setInt = new HashSet<>();
                    for (String s : inputList) {
                        setInt.add(Integer.parseInt(s));
                    }
                    return setInt;
                } catch (NumberFormatException e) {
                    System.out.println("Error: set can only consist of integers");
                }
            }
        }
    }

    public static void main(String[] args) {
        System.out.println("Enter set A:");
        Set<Integer> setA = inputSet();

        System.out.println("Enter set B:");
        Set<Integer> setB = inputSet();

        System.out.println("Enter set U:");
        Set<Integer> setU = inputSet();

        System.out.println("Enter set C:");
        Set<Integer> setC = inputSet();

        Set<Integer> result = new HashSet<>(setA);
        result.retainAll(setB);
        Set<Integer> setUC = new HashSet<>(setU);
        setUC.removeAll(setC);
        result.addAll(setUC);

        System.out.println("Transformed set: " + result);
    }
}

File: Fundamentals of Discrete mathematics/C++/task_1.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <set>

using namespace std;

set<int> input_set() {
    while (true) {
        string input;
        getline(cin, input);
        stringstream ss(input);
        set<int> s;
        int x;
        while (ss >> x) {
            s.insert(x);
        }
        if (ss.eof() && !ss.fail()) {
            return s;
        } else {
            cout << "Error: set can only consist of integers" << endl;
        }
    }
}

int main() {
    cout << "Enter set A:" << endl;
    set<int> set_a = input_set();

    cout << "Enter set B:" << endl;
    set<int> set_b = input_set();

    cout << "Enter set U:" << endl;
    set<int> set_u = input_set();

    cout << "Enter set C:" << endl;
    set<int> set_c = input_set();

    set<int> result;
    set_intersection(set_a.begin(), set_a.end(), set_b.begin(), set_b.end(), inserter(result, result.begin()));
    set_difference(set_u.begin(), set_u.end(), set_c.begin(), set_c.end(), inserter(result, result.begin()));
    cout << "Transformed set: ";
    for (int x : result) {
        cout << x << " ";
    }
    cout << endl;

    return 0;
}
